com.unity.ml-agents.extensions/Runtime/Sensors/ArticulationBodyPoseExtractor.cs

```diff
@@ -38,23 +43,35 @@ public ArticulationBodyPoseExtractor(ArticulationBody rootBody)
 
             for (var i = 1; i < numBodies; i++)
             {
-                var body = m_Bodies[i];
-                var parent = body.GetComponentInParent<ArticulationBody>();
-                parentIndices[i] = bodyToIndex[parent];
+                var currentArticBody = m_Bodies[i];
+                // Component.GetComponentInParent will consider the provided object as well.
+                // So start looking from the parent.
+                var currentGameObject = currentArticBody.gameObject;
+                var parentGameObject = currentGameObject.transform.parent;
+                var parentArticBody = parentGameObject.GetComponentInParent<ArticulationBody>();
+                parentIndices[i] = bodyToIndex[parentArticBody];
             }
 
             SetParentIndices(parentIndices);
         }
 
+        protected ArticulationBody GetBodyAt(int index)
+        {
+            return m_Bodies[index];
+        }
+
+        protected override Vector3 GetLinearVelocityAt(int index)
+        {
+            return GetBodyAt(index).velocity;
```

Comment:
any reason to call `GetBodyAt` here, but not line 70?  Is `GetBodyAt` expected to be overridden, or can you just use the array accessor here?

Reply:
Good call, I'm going to get rid of this (and same in the Rigidbody version)

Reply:
Updated.